copy/copy.go

```diff
@@ -257,6 +297,10 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 		return nil, errors.Wrapf(err, "Error initializing image from source %s", transports.ImageName(c.rawSource.Reference()))
 	}
 
+	if options.EncryptLayers != nil && !src.SupportsEncryption(ctx) {
```

Comment:
Separately, there should exist something to reject copying encrypted blobs (whether unmodified, or because the caller requests encryption), to transports / formats that can’t support them.

That’s at least `containers-storage:`, and there may be others (at least the `docker/tarfile` two).

Reply:
Added this in https://github.com/containers/image/pull/673/commits/78c7bece4c6f6e977aec63b832b0d0adf2e6e9b4